android/src/meez/nativeExtensions/androidyoutube/AndroidYoutubeContext.java

```diff
@@ -168,21 +167,55 @@ public void dispose()
      */
     public void initVideo(String devKey)
     {
-        Log.d(Extension.TAG, "AndroidYouTubeContext.initVid()");
+        Log.d(Extension.TAG, "AndroidYouTubeContext.initVideo()");
+
+        assertState(State.INITIALIZING);
+
+        // Create views
+        this.videoContainer = createVideoContainer();
+        this.videoContainer.setId(AndroidYoutubeContext.generateViewId());
+        this.layoutParams = createLayoutParams();
+        this.dialog = createDialog();
+        this.playerFragment = createPlayerFragment(new FragmentViewListener(){
+            @Override public void  onFragmentViewCreated() {
+                if (checkState(State.DISPOSING) || checkState(State.DISPOSED))
+                    return;
+
+                // Remove from root and add to dialog
+                getRootContainer().removeView(videoContainer);
 
-        //Hack - do not want to see video container until it has been added to dialog
-        getVideoContainer().setVisibility(View.INVISIBLE);
+                // hide video offscreen until explicitly positioned by Actionscript call
+                setVideoFrame(-5000, -5000, 600, 400);
+
+                dialog.setContentView(videoContainer);
+                dialog.show();
+
+                videoContainer.setVisibility(View.VISIBLE);
+
+                // Leave init state when view is ready.
+                changeState(State.STOPPED);
+            }
+            @Override public void onSaveInstance(boolean saveState){
+                if (checkState(State.DISPOSING) || checkState(State.DISPOSED))
+                    return;
+                setVideoSaveEnabled(saveState);
+            }
+        });
 
-        //Hack - videoContainer is only added to view long enough to attach fragment.
+        //Hack - videoContainer is only added to root view long enough to attach fragment (fragment must be attached to view in Android 'display list').
```

Comment:
The dialog was added as I didn't know a way of absolutely positioning a FrameLayout (or a fragment within a FrameLayout). I am open to any ideas to get rid of that.